engine/runtime/src/memories/frag.rs

```diff
@@ -0,0 +1,624 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides functionality to allocate memory with higher randomness on returned addresses.
+//!
+//! The most simple approach is to allocate memory multiple times, return the final allocation
+//! as the desired memory.
+//!
+//! Allocators differ between operating systems. On *nix and BSD-based systems `malloc(int)` is being called
+//! but does not initialize the allocated spaced. Microsoft Windows uses
+//! [`VirtualAlloc`](https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualalloc)
+//! allocates and initializes a region of memory pages with zeroes.
+//!
+//! [`FragStrategy`] implements at least two possible allocation strategies:
+//!
+//! - Direct: The algorithm tries to allocate a huge amount of memory space while keeping a certain address distance
+//! - Memory Mapped: anonymous memory is being mapping, the memory address will be randomly selected.
+
+use crate::MemoryError;
+use log::*;
+use std::{fmt::Debug, ptr::NonNull};
+use zeroize::Zeroize;
+
+// The minimum distance we consider between 2 fragments
+// This is the page size for most linux systems
+pub const FRAG_MIN_DISTANCE: usize = 0x1000;
```

Comment:
yeah, that's linux "default" page size, I wonder if we should keep this somewhat externally configurable :thinking:

Reply:
I can make it a static variable that is initialized once with system page size or with a default value if we can't get it